### PR TITLE
Add support for Portage, XBPS and Nix to setupbash.sh

### PR DESCRIPTION
**Pull request by AlbydST**
Added support for Gentoo's Portage, Void's XBPS and Nix to setupbash.sh

### CODE REVIEW

File: src/commands/setupbash.sh [hunk 1]
```diff
@@ -64,11 +64,11 @@ installDepend() {
     if [[ $PACKAGER == "pacman" ]]; then
         if ! command_exists yay && ! command_exists paru; then
             echo "Installing yay as AUR helper..."
-            sudo ${PACKAGER} --noconfirm -S base-devel
+            ${PACKAGER} --noconfirm -S base-devel
```
Comment (4yman-el) [inline src/commands/setupbash.sh:67]:
```suggestion
            sudo ${PACKAGER} --noconfirm -S base-devel
```

We still need to use sudo

File: src/commands/setupbash.sh [hunk 2]
```diff
@@ -79,8 +79,14 @@ installDepend() {
...
         ${AUR_HELPER} --noconfirm -S ${DEPENDENCIES}
+    elif [[ $PACKAGER == "emerge" ]]; then
+        ${PACKAGER} -v app-shells/bash app-shells/bash-completion app-arch/tar app-editors/neovim sys-apps/bat app-text/tree app-text/multitail app-misc/fastfetch
+    elif [[ $PACKAGER == "xbps-install" ]]; then
+        ${PACKAGER} -v ${DEPENDENCIES}
+    elif [[ $PACKAGER == "nix-env" ]]; then
+        ${PACKAGER} -iA nixos.bash nixos.bash-completion nixos.gnutar nixos.neovim nixos.bat nixos.tree nixos.multitail nixos.fastfetch
     else
-        sudo ${PACKAGER} install -yq ${DEPENDENCIES}
+        ${PACKAGER} install -yq ${DEPENDENCIES}
```
Comment (4yman-el) [inline src/commands/setupbash.sh:89]:
```suggestion
        sudo ${PACKAGER} install -yq ${DEPENDENCIES}
```

**Review by 4yman-el — CHANGES_REQUESTED** [review]

**Comment by AlbydST** [comment]
I've made it so it supports both doas and sudo, and if neither is present it just runs the command with su -c

**Comment by ChrisTitusTech** [comment]
Love the PR @AlbydST but I had to move this to the mybash project and change the code a bit. I put it in this commit e9382f6b861824ec0ed1d233113ebdd90ea17b9e

**Pull request closed without merging**